### PR TITLE
Fix modal verb "must"

### PR DESCRIPTION
**Pull request by burblebee**
Partially addresses cplusplus/nbballot#393

### CODE REVIEW

File: source/containers.tex [hunk 1]
```diff
@@ -4079,10 +4079,12 @@
...
 Modifying any list requires access to the element preceding the first element
 of interest, but in a \tcode{forward_list} there is no constant-time way to access a
-preceding element. For this reason, ranges that are modified, such as those supplied to
-\tcode{erase} and \tcode{splice}, must be open at the beginning.
+preceding element.
+For this reason, \tcode{erase_after} and \tcode{splice_after}
+take fully-open ranges, not semi-open ranges.
 \end{note}
 
+%FIXME: Are these examples wrong?
```
Comment (zygoloid) [inline source/containers.tex:4087]: It looks like this FIXME is applying to the `forward_list` synopsis. Is the FIXME still necessary after the above change?
```suggestion
```

File: source/expressions.tex [hunk 2]
```diff
@@ -6612,7 +6615,7 @@
 no new exception object is created.
 The exception is no longer considered to be caught.
 \begin{example}
-Code that must be executed because of an exception, but cannot
+Code that is executed because of an exception, but cannot
 completely handle the exception itself, can be written like this:
```
Comment (zygoloid) [inline source/expressions.tex:6619]: I don't like this phrasing (before or after the "must" fix). I think this would be good enough for now:
```suggestion
An exception handler that cannot completely handle the exception itself
can be written like this:
```

File: source/grammar.tex [hunk 3]
```diff
@@ -10,8 +10,8 @@
 In particular, the grammar described here accepts
 a superset of valid \Cpp{} constructs.
 Disambiguation rules~(\ref{stmt.ambig}, \ref{dcl.spec}, \ref{class.member.lookup})
-must be applied to distinguish expressions from declarations.
-Further, access control, ambiguity, and type rules must be used
+shall be applied to distinguish expressions from declarations.
+Further, access control, ambiguity, and type rules shall be used
```
Comment (zygoloid) [inline source/grammar.tex:14]:
```suggestion
are applied to distinguish expressions from declarations.
Further, access control, ambiguity, and type rules are used
```

We can't "shall" from an informative Annex, and I think this is supposed to be descriptive rather than introducing a requirement.

File: source/iostreams.tex [hunk 4]
```diff
@@ -14251,8 +14249,7 @@
 \begin{note}
 Implementations of \tcode{directory_iterator}\iref{fs.class.directory.iterator}
 are prohibited from directly or indirectly calling the \tcode{refresh} function
-since it must access the external file system,
-and the objective of caching is to avoid unnecessary file system accesses.
+as described in \ref{fs.class.directory.iterator.general}".
```
Comment (zygoloid) [inline source/iostreams.tex:14252]:
```suggestion
as described in \ref{fs.class.directory.iterator.general}.
```

File: source/overloading.tex [hunk 5]
```diff
@@ -1604,7 +1604,7 @@
 \pnum
 When resolving a placeholder for a deduced class type\iref{dcl.type.simple}
 where the \grammarterm{template-name} names an alias template \tcode{A},
-the \grammarterm{defining-type-id} of \tcode{A} must be of the form
+the \grammarterm{defining-type-id} of \tcode{A} shall be of the form
```
Comment (zygoloid) [inline source/overloading.tex:1607]: We have an "as specified in" below pointing to a normative rule elsewhere, so I think "must" here was OK following our "internal constraint" meaning.
```suggestion
the \grammarterm{defining-type-id} of \tcode{A} must be of the form
```

File: source/statements.tex [hunk 6]
```diff
@@ -1001,8 +1001,9 @@
 declaration.
 If control enters the declaration concurrently while the variable is
 being initialized, the concurrent execution shall wait for completion
-of the initialization.\footnote{The implementation must not introduce
-any deadlock around execution of the initializer. Deadlocks might
+of the initialization. The implementation shall not introduce
+any deadlock around execution of the initializer.
+\footnote{Deadlocks might
 still be caused by the program logic; the implementation need only
 avoid deadlocks due to its own synchronization operations.} If control
```
Comment (zygoloid) [inline source/statements.tex:1008]: I'm nervous about this; the phrasing of this is vague, and "shall"ing it might add a requirement that's stronger than what we meant. I think rather than a "shall" here, we want to point out that introducing a deadlock would not satisfy "as if". How about this:

```suggestion
of the initialization.
\begin{note}
A conforming implementation cannot introduce
any deadlock around execution of the initializer.
Deadlocks might still be caused by the program logic;
the implementation need only avoid deadlocks
due to its own synchronization operations.
\end{note}
If control
```

File: source/templates.tex [hunk 7]
```diff
@@ -694,18 +694,13 @@
...
-For a
-\grammarterm{template-name}
-to be explicitly qualified by the template arguments,
-the name must be considered to refer to a template.
+An identifier is a \grammarterm{template-name}
+if it is associated by name lookup with
+a template or an overload set that contains a function template, or,
+if the identifier is followed by \tcode{<},
+the \grammarterm{template-id} would form an \grammarterm{unqualified-id}, and
+name lookup either finds one or more functions or finds nothing.
```
Comment (zygoloid) [inline source/templates.tex:703]: Please also delete the text after the note that this partially duplicates.

File: source/templates.tex [hunk 8]
```diff
@@ -694,18 +694,13 @@
...
-that is, to identify a name as a
-\grammarterm{template-name}.
-\end{note}
-
 \pnum
-For a
-\grammarterm{template-name}
-to be explicitly qualified by the template arguments,
-the name must be considered to refer to a template.
+An identifier is a \grammarterm{template-name}
```
Comment (zygoloid) [inline source/templates.tex:698]:
```suggestion
An \grammarterm{identifier} is a \grammarterm{template-name}
```

File: source/templates.tex [hunk 9]
```diff
@@ -694,18 +694,13 @@
...
+if it is associated by name lookup with
+a template or an overload set that contains a function template, or,
+if the identifier is followed by \tcode{<},
```
Comment (zygoloid) [inline source/templates.tex:701]:
```suggestion
a template or an overload set that contains a function template, or
the \grammarterm{identifier} is followed by \tcode{<},
```

File: source/templates.tex [hunk 10]
```diff
@@ -7136,12 +7131,13 @@
 otherwise, it is a declaration.
 \begin{note}
 The definition of a static data member of a template
-that requires default-initialization must use a \grammarterm{braced-init-list}:
+that requires default-initialization
```
Comment (zygoloid) [inline source/templates.tex:7134]:
```suggestion
for which default-initialization is desired
```
(Just a drive-by to avoid using "requires" where we're not specifying a requirement.)

File: source/threads.tex [hunk 11]
```diff
@@ -94,7 +94,7 @@
 The functions whose names end in \tcode{_for} take an argument that
 specifies a duration. These functions produce relative timeouts. Implementations
 should use a steady clock to measure time for these functions.\footnote{All
```
Comment (zygoloid) [inline source/threads.tex:96]:
```suggestion
should use a steady clock to measure time for these functions.%
```

File: source/threads.tex [hunk 12]
```diff
@@ -94,7 +94,7 @@
...
-implementations for which standard time units are meaningful must necessarily
+implementations for which standard time units are meaningful will typically
```
Comment (zygoloid) [inline source/threads.tex:97]:
```suggestion
\footnote{Implementations for which standard time units are meaningful will typically
```
("All implementations [...] will typically" sounds self-contradictory.)

**Review by zygoloid — COMMENTED** [review]
Please include "Partially addresses ISO/CS 017 (C++20 DIS)." in the commit description.

File: source/containers.tex (hunk 1 above)
Comment (burblebee) [reply source/containers.tex:4087]: Removed FIXME

File: source/expressions.tex (hunk 2 above)
Comment (burblebee) [reply source/expressions.tex:6619]: done

File: source/grammar.tex (hunk 3 above)
Comment (burblebee) [reply source/grammar.tex:14]: done

File: source/iostreams.tex (hunk 4 above)
Comment (burblebee) [reply source/iostreams.tex:14252]: doh!  thanks, done

File: source/overloading.tex (hunk 5 above)
Comment (burblebee) [reply source/overloading.tex:1607]: done

File: source/statements.tex (hunk 6 above)
Comment (burblebee) [reply source/statements.tex:1008]: done

File: source/templates.tex (hunk 8 above)
Comment (burblebee) [reply source/templates.tex:698]: I initially grammarterm'ed these identifiers, but we never talk about a grammarterm being another grammarterm; we speak of it as having the form of another grammarterm.

Suggestions applied nonetheless.

File: source/threads.tex (hunk 11 above)
Comment (burblebee) [reply source/threads.tex:96]: done, but why the trailing %?

Comment (zygoloid) [reply source/threads.tex:96]: To avoid a space between the period and the footnote marker.

**Comment by burblebee** [comment]
> Please include "Partially addresses ISO/CS 017 (C++20 DIS)." in the commit description.

Will do when I rebase and force push; should I do that now or wait for more feedback?

**Comment by burblebee** [comment]
> > Please include "Partially addresses ISO/CS 017 (C++20 DIS)." in the commit description.
> 
> Will do when I rebase and force push; should I do that now or wait for more feedback?

Went ahead and did this now since c++20 was force-pushed.

File: source/templates.tex [hunk 13]
```diff
@@ -715,10 +710,6 @@
 name lookup finds
 a \grammarterm{template-name}
 or an overload set that contains a function template.
```
Comment (zygoloid) [inline source/templates.tex:712]: Please remove this sentence too, this is also captured by the new text above.
